### PR TITLE
Use a real anchor for remote page attachments

### PR DESCRIPTION
**Pull request by cramforce**
...which are really just links.

Also:

- Changes vertical rendering to not specify a height for attachments such that they simply flow as high as their content.
- Fixes menu overlaying all vertically rendered content making it non-interactive.

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-share-menu.css [hunk 1]
```diff
@@ -26,11 +26,13 @@
   z-index: 100003 !important; /** Above pagination-buttons. */
   transform: translate3d(0, 100vh, 0) !important;
   transition-delay: 0.15s !important;
+  pointer-events: none;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-share-menu.css:29]:
```suggestion
  pointer-events: none !important;
```

File: extensions/amp-story/1.0/amp-story-share-menu.css [hunk 2]
```diff
@@ -26,11 +26,13 @@
...
 }
 
 .i-amphtml-story-share-menu-visible {
   transform: translate3d(0, 0, 0) !important;
   transition-delay: 0s !important;
+  pointer-events: auto;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-share-menu.css:35]:
```suggestion
  pointer-events: auto !important;
```

**Comment by cramforce** [comment]
@gmajoulet Can you explain why Percy did not report a change? I changed the visual test.

**Comment by gmajoulet** [comment]
Good catch, the test was marked as flaky in https://github.com/ampproject/amphtml/issues/27452, fixed in https://github.com/ampproject/amphtml/issues/27185 but not re-enabled.

Can you enable it back?

https://github.com/ampproject/amphtml/blob/1b25eee74f70a63dcce39c309ab9d5fc30a6e0c4/test/visual-diff/visual-tests#L659-L668

File: extensions/amp-story/1.0/amp-story-page-attachment.js [hunk 3]
```diff
@@ -146,13 +146,14 @@ export class AmpStoryPageAttachment extends DraggableDrawer {
...
-            <span class="i-amphtml-story-page-attachment-remote-icon"></span>
-          </div>`
-    );
+    // Use an anchor element to make this a real link in vertical rendering.
+    const link = htmlFor(this.element)`
+    <a class="i-amphtml-story-page-attachment-remote-content" target="_blank">
+      <span class="i-amphtml-story-page-attachment-remote-domain"></span>
+      <span class="i-amphtml-story-page-attachment-remote-icon"></span>
+    </a>`;
+    link.setAttribute('href', this.element.getAttribute('href'));
```
Comment (kristoferbaxter) [inline extensions/amp-story/1.0/amp-story-page-attachment.js:155]: Is it not possible to set this href in the `htmlFor` template?

```javascript
htmlFor(thisElement)`<a href=${this.element.getAttribute('href')}>...</a>`
```

Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story-page-attachment.js]: You can't do that, `htmlFor` templates don't accept variables :(

Comment (kristoferbaxter) [reply extensions/amp-story/1.0/amp-story-page-attachment.js:155]: :( Sorry for the distraction.

I'll look into that separately, that's a super bummer.

**Pull request merged**